### ISSUE TITLE
Problem accounting for different status bar + notch sizes across different phones

### ISSUE THREAD

**Issue by BrylleGomez** [OpenFlutter/flutter_screenutil#213]
[Hi!](url) First of all, thank you for such an amazing library! It works perfectly when I am not dealing with any appbars or status bars (when I am designing for full screen). However, when I finally need to take into account status bar + app bar, the library behaves unpredictably. 

For example, for a design blueprint (iPhone X on Figma) screen size of (375, 812), I have a Column of components as follows:
- Status bar + notch --> **44.h** from original blueprint (but variable for each phone)
- Appbar (preferred size) --> **44.h**
- Box --> **96.h**
- Another box --> **628.h**
- Total = 44 + 44 + 96 + 628 = 812

Now for my Pixel 3 emulator, setting status bar + notch to 14.h does the trick. However for other phones this does not work. I am not quite sure how to account for the differing status bar + notch heights. I am also not sure how to make use of ScreenUtil().statusBarHeight, as I have tried somehow including it in the equation but can't seem to make it work. Can you please assist me on this? Do let me know if you need more info. Thanks! (Apologies for my lack of understanding on how to use your library if ever) 

![image](https://user-images.githubusercontent.com/43936757/107129004-944af900-68db-11eb-9703-84a2201a18a7.png)

**Comment by lizhuoyuan** [comment]
现在， 最新版本初始化已经改成单位为dp了，不过因为影响较大可以依然使用px，只是不确定版本有没有对像素密度做处理。
以下内容基于最新版：
另外，获取状态栏高度的方法，拿到的值单位是pd，获取屏幕的高度单位也为dp，如果不确定当前使用的版本的行为，可以查看对应方法的注释。
另外，官方有获取状态栏高度以及屏幕高度的方法:
```
    MediaQuery.of(context).padding.top

 MediaQuery.of(context).size.height

```

**Comment by BrylleGomez** [comment]
Thank you for the swift reply. I've actually discovered a formula for calculating the proper height of the app bar given different status bar sizes. 

a' = app bar height in code (in dp)
a = app bar height in blueprint (in dp)
s' = status bar height in code (in dp), retrievable by ScreenUtil().statusBarHeight
s = status bar height in bueprint (in dp)

Magic formula is --> **_a' + s' = (a + s).h_** 
So I used _a' = (a + s).h - s'_ to get the proper app bar height that changes based on status bar of any phone and maintains the total number of pixels in the height. In code, this is how it looks:

```
appBar: PreferredSize(
        preferredSize: Size(375.w, (88.h - ScreenUtil().statusBarHeight))  
// 88.h is the combined height of app bar and status bar in the iPhone X blueprint (44.h + 44.h)
```

Just sharing this in case it might be helpful to everyone. Also not sure whether you @lizhuoyuan intended this magic formula but it's there and it works like a charm. Again, thank you for your great work on the library!
